src/ripple/app/ledger/LedgerConsensus.h

```diff
@@ -38,26 +39,36 @@ namespace ripple {
     This object is created when the consensus process starts, and
     is destroyed when the process is complete.
 */
-class LedgerConsensus
+template <class Traits>
+class LedgerConsensus : public Traits
 {
 public:
+
+    using typename Traits::CxTime_t;
+    using typename Traits::CxPos_t;
+    using typename Traits::CxTxSet_t;
+    using typename Traits::CxTx_t;
+    using typename Traits::CxLgrID_t;
+    using typename Traits::CxTxID_t;
+    using typename Traits::CxTxSetID_t;
+    using typename Traits::CxNodeID_t;
+
     virtual ~LedgerConsensus() = default;
 
     virtual Json::Value getJson (bool full) = 0;
 
-    virtual uint256 getLCL () = 0;
+    virtual CxLgrID_t getLCL () = 0;
```

Comment:
Do I understand this right, the base class `LedgerConsensus` implements the pure algorithm, while the derived class provides the application specific behaviors (e.g. RCL)? If so, we should use CRTP not pure virtuals.